packages/common-sdk/src/web3/transactions/transactions-processor.ts

```diff
@@ -59,47 +58,57 @@ export class TransactionProcessor {
   public constructSendTransactions(
     transactions: Transaction[],
     lastValidBlockHeight: number,
+    blockhash: string,
     parallel: boolean = true
   ): () => Promise<PromiseSettledResult<string>[]> {
-    return async () => {
-      let done = false;
-      const isDone = () => done;
+    const executeTx = async (tx: Transaction) => {
+      const rawTxs = tx.serialize();
+      return this.connection.sendRawTransaction(rawTxs, {
+        preflightCommitment: this.commitment,
+      });
+    };
 
-      // We separate the block expiry promise so that it can be shared for all the transactions
-      const expiry = checkBlockHeightExpiry(
-        this.connection,
-        lastValidBlockHeight,
-        this.commitment,
-        isDone
-      );
-      const txs = transactions.map((tx) => tx.serialize());
-      const txPromises = txs.map(async (tx) =>
-        confirmOrExpire(this.connection, tx, this.commitment, expiry)
-      );
-      let results: PromiseSettledResult<string>[] = [];
+    const confirmTx = async (txId: string) => {
+      const result = await this.connection.confirmTransaction({
+        signature: txId,
+        lastValidBlockHeight: lastValidBlockHeight,
+        blockhash,
+      });
+
+      if (result.value.err) {
```

Comment:
note that our transaction confirmation promise now rejects if the program fails (e.g. slippage failure). this is different from the previous behavior, where the promise did not reject, and we showed the "Transaction succeeded" snackbar

imo, this new implementation is not ideal. the ideal is that the function immediately returns the txid after sending the transaction but before confirming the transaction. this allows the UI to show a pending snackbar. afterwards, the txid should be used to separately confirm the transaction

however, this would require a deeper change in `TransactionProcessor`, which i think we can reserve for our future refactor